test/pylib/artifact_registry.py

```diff
@@ -32,8 +32,6 @@ def __init__(self) -> None:
     async def cleanup_before_exit(self) -> None:
         logging.info("Cleaning up before exit...")
         for artifacts in self.suite_artifacts.values():
-            for artifact in artifacts:
-                artifact.close()  # type: ignore
```

Comment:
A python future has  a close method, but it's not documented. It cancel the future from being executed by the reactor. Your comment is wrong - if there is an internal error in test.py this code is called.

Reply:
Turns out the correct type for `artifact` here is `Coroutine`. It's a subtype of `Awaitable` which has a `close()` method.
https://docs.python.org/3/reference/datamodel.html#coroutine-objects